### PR TITLE
Truncate property key and value if it is longer than limit

### PR DESCRIPTION
**Pull request by jaelim-ms**

### CODE REVIEW

File: sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java [hunk 1]
```diff
@@ -277,36 +288,36 @@ private static boolean validateName(String name, String logType) {
...
-                message = String.format("%s '%s' : property '%s' : property key length cannot be longer than %s characters. Property '%s' will be skipped.", logType, logName, property.getKey(), maxPropertyItemLength, property.getKey());
+            if (value == null) {
+                message = String.format("%s '%s' : property '%s' : property value cannot be null. Property '%s' will be skipped.", logType, logName, key, key);
                 MobileCenterLog.warn(Analytics.LOG_TAG, message);
                 continue;
             }
-            if (property.getValue() == null) {
-                message = String.format("%s '%s' : property '%s' : property value cannot be null. Property '%s' will be skipped.", logType, logName, property.getKey(), property.getKey());
+            if (key.length() > MAX_PROPERTY_ITEM_LENGTH) {
+                message = String.format("%s '%s' : property '%s' : property key length cannot be longer than %s characters. Property key '%s' will be truncated.", logType, logName, key, MAX_PROPERTY_ITEM_LENGTH, key);
```
Comment (guperrot) [inline sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java]: Message is slighty different than already merged https://github.com/Microsoft/mobile-center-sdk-dotnet/pull/391/files  

"Property key will be truncated." at the end does not have '%s' in c# and should be the same in Android.

File: sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java [hunk 2]
```diff
@@ -277,36 +288,36 @@ private static boolean validateName(String name, String logType) {
...
                 MobileCenterLog.warn(Analytics.LOG_TAG, message);
-                continue;
+                key = key.substring(0, MAX_PROPERTY_ITEM_LENGTH);
             }
-            if (property.getValue().length() > maxPropertyItemLength) {
-                message = String.format("%s '%s' : property '%s' : property value cannot be longer than %s characters. Property '%s' will be skipped.", logType, logName, property.getKey(), maxPropertyItemLength, property.getKey());
+            if (value.length() > MAX_PROPERTY_ITEM_LENGTH) {
+                message = String.format("%s '%s' : property '%s' : property value cannot be longer than %s characters. Property '%s' will be truncated.", logType, logName, key, MAX_PROPERTY_ITEM_LENGTH, value);
```
Comment (guperrot) [inline sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java]: same comment as key

File: sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java (hunk 1 above)
Comment (jaelim-ms) [inline sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java]: Name too? I didn't know about that. Will add that to the PR.

Comment (guperrot) [inline sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java]: It was not in original tfs but I confirmed with @elamalani. But this comment you reply too is a different comment about message to print for event key and value (to align with .net)

File: sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java [hunk 3]
```diff
@@ -250,19 +271,18 @@ private static String generatePageName(Class<?> activityClass) {
...
         if (name == null || name.isEmpty()) {
             MobileCenterLog.error(Analytics.LOG_TAG, logType + " name cannot be null or empty.");
-            return false;
+            return null;
         }
-        if (name.length() > maxNameLength) {
-            MobileCenterLog.error(Analytics.LOG_TAG, String.format("%s '%s' : name length cannot be longer than %s characters.", logType, name, maxNameLength));
-            return false;
+        if (name.length() > MAX_NAME_LENGTH) {
+            MobileCenterLog.error(Analytics.LOG_TAG, String.format("%s '%s' : name length cannot be longer than %s characters. Name will be truncated.", logType, name, MAX_NAME_LENGTH));
```
Comment (guperrot) [inline sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java]: Use warnings whenever we truncate, not an error.

**Pull request merged**